File: backend/cs_api/src/Application/Dtos/ExchangeRatesDto.cs
namespace cs_api.src.Application.Dtos
{
  public class UsdRatesDto
  {
    // Brazilian Reais
    public decimal Brl { get; set; }
    // Euro
    public decimal Eur { get; set; }
    // Canadian dollar
    public decimal Cad { get; set; }
    // Mexican Pesos
    public decimal Mxn { get; set; }
    // Libras
    public decimal Gbp { get; set; }
  }
  public class ExchangeRatesDto
  {
    public DateTime? Date { get; set; }
    public UsdRatesDto? Rates { get; set; }
  }
}

File: backend/_others_/node/src/exchange-rate/dto/exchange-rates.dto.ts
export interface IUsdRatesDto {
  brl: number; // Brazilian Reais
  eur: number; // Euro
  cad: number; // Canadian dollar
  mxn: number; // Mexican Pesos
  gbp: number; // Libras
}

export interface IExchangeRatesDto {
  date: string;
  usd: IUsdRatesDto;
}
